core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +167,68 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              updateSearchFieldsBasedOnUrlQuery();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var updateLanguageCodeFields = function(url) {
+          // Grabs Grab language code(s) from URL and returns url without
+          // language fields. This function assumes that language codes are
+          // at the end of the url.
+
+          var _languageCodePattern = / language_code=\([a-zOR\%\" ]+\)$/;
+          var _languageCodes = url.match(_languageCodePattern)[0];
+          var _languageCodeSelections = [];
+          if (_languageCodes != undefined) {
+            url = url.replace(_languageCodePattern, '');
```

Comment:
Don't rewrite 'url'. Just extract the languageCode bit from it, using a separate variable if necessary. It's not good for a variable to mean multiple different things in the lifecycle of a single function.